File: algorithms/interview-preparation-kit-2/js/code/pokemonCollection/pokemonCollection.spec.mjs
import { pokemonCollection } from './pokemonCollection.mjs';

describe('Pokemon Collection', () => {
  test.skip('collection with 1 pokemon not including repetitions', () => {
    const pokemons = ['Zubat'];
    expect(pokemonCollection(pokemons)).toBe(150);
  });

  test.skip('collection with 1 pokemon including repetitions', () => {
    const pokemons = ['Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat'];
    expect(pokemonCollection(pokemons)).toBe(150);
  });

  test.skip('collection with 2 pokemon including repetitions', () => {
    const pokemons = ['Zubat', 'Pikachu', 'Pikachu'];
    expect(pokemonCollection(pokemons)).toBe(149);
  });

  test.skip('collection with 4 pokemon not including repetitions', () => {
    const pokemons = ['Zubat', 'Charmander', 'Caterpie', 'Pidgeot'];
    expect(pokemonCollection(pokemons)).toBe(147);
  });

  test.skip('collection with 5 pokemon including repetitions', () => {
    const pokemons = ['Charmander', 'Caterpie', 'Pidgeot', 'Rattata', 'Zubat', 'Zubat', 'Zubat'];
    expect(pokemonCollection(pokemons)).toBe(146);
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code/pokemonCollection/pokemonCollection.spec.rb
require_relative "./pokemonCollection"

describe "Pokemon Collection" do
  context "collection with 1 pokemon not including repetitions" do
    it "it 150" do
      pokemons = ['Zubat'];
      expect(pokemon_collection(pokemons)).to eq(150)
    end
  end    
end

describe "Pokemon Collection" do
  context "collection with 1 pokemon including repetitions" do
    it "it 150" do
      pokemons = ['Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat'];
      expect(pokemon_collection(pokemons)).to eq(150)
    end
  end    
end

describe "Pokemon Collection" do
  context "collection with 2 pokemon including repetitions" do
    it "it 149" do
      pokemons = ['Zubat', 'Pikachu', 'Pikachu'];
      expect(pokemon_collection(pokemons)).to eq(149)
    end
  end    
end

describe "Pokemon Collection" do
  context "collection with 4 pokemon not including repetitions" do
    it "it 147" do
      pokemons = ['Zubat', 'Charmander', 'Caterpie', 'Pidgeot'];
      expect(pokemon_collection(pokemons)).to eq(147)
    end
  end    
end

describe "Pokemon Collection" do
  context "collection with 5 pokemon including repetitions" do
    it "it 146" do
      pokemons = ['Charmander', 'Caterpie', 'Pidgeot', 'Rattata', 'Zubat', 'Zubat', 'Zubat'];
      expect(pokemon_collection(pokemons)).to eq(146)
    end
  end    
end

